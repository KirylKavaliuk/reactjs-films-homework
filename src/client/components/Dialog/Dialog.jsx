import React, { Component } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import styles from './Dialog.scss';

export default class Dialog extends Component {
  render() {
    return (
      <div className={
        classNames(
          styles.dialog,
          { [styles.open]: this.props.open },
        )}
        onClick={ this.props.closeDialog }
      >
        { this.props.component || <div>dialog</div> }
      </div>
    );
  }
}

Dialog.defaultProps = {

};

Dialog.propTypes = {

};
