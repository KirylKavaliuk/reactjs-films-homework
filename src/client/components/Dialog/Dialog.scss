.dialog {
  position: fixed;
  z-index: 200;
  top: 0;
  left: 0;
  display: none;
  overflow: hidden;
  width: 100%;
  min-height: 100vh;
  background: rgba(0, 0, 0, 0.7);
}

.open {
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
}
