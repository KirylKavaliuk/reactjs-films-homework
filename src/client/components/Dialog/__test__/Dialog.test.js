import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Dialog from '../Dialog';

const Component = () => (<div>component for dialog</div>);

it('<Dialog/>', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Dialog open>
          <Component/>
        </Dialog>
      </MemoryRouter>,
    ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<Dialog/>', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Dialog/>
      </MemoryRouter>,
    ).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});
