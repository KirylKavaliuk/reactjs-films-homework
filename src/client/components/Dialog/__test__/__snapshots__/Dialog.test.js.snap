// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog/> 1`] = `
<a
  className="Link__link___-IU0S Dialog__dialog___3Z4Fy Dialog__open___oVAzw"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div>
    component for dialog
  </div>
</a>
`;

exports[`<Dialog/> 2`] = `null`;
